flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/AggregateUtil.scala

```diff
@@ -290,6 +295,10 @@ object AggregateUtil {
     val aggMapping = aggregates.indices.toArray.map(_ + groupings.length)
     val outputArity = aggregates.length + groupings.length + 1
 
+    // remove when distinct is supported
+    val distinctAggregatesFlags = new Array[Boolean](aggregates.size)
```

Comment:
`DataSet` has it's own way to deal with distinct aggregates. They could also not initialize the state. So it is save to call `generateAggregations()` with an empty array: `Array[Boolean]()` or we make the parameter an `Option[Array[Boolean]]` and pass `None`